inc/cxplat.hpp

```diff
@@ -77,17 +65,67 @@ class CxPlatAsync {
 #if defined(CX_PLATFORM_WINUSER) || defined(CX_PLATFORM_WINKERNEL)
     bool WaitFor(uint32_t TimeoutMs) noexcept {
         if (Initialized) {
-            if (CxPlatThreadWaitWithTimeout(&Thread, TimeoutMs)) {
-                return true;
-            }
+            return (ThreadCompleted = CxPlatThreadWaitWithTimeout(&Thread, TimeoutMs));
         }
         return false;
     }
 #endif
 
-    void* Get() noexcept {
+    R Get() noexcept {
         return AsyncContext.ReturnValue;
     }
 };
 
+template <typename T>
```

Comment:
Would it be better to always define the callback without a return value so that we don't have to dip this class?

Without supporting lambda capture, this return value doesn't help much because the caller needs to define a context struct to pass all local vars into the callback. The caller could simply add one more field for the return value in the context struct.

Reply:
I don't have strong feelings either way. The code is written already; and works. There are instances in XDP which are using it too. If I had to choose, I'd argue we keep it like this.

Reply:
I think we should remove the return value entirely. I can update the xdp code after this PR is merged.

Reply:
I'll go ahead and do it now real quick.

Reply:
Done